container_linux.go

```diff
@@ -169,21 +168,58 @@ func (c *linuxContainer) newInitProcess(p *Process, cmd *exec.Cmd, parentPipe, c
 			cmd.SysProcAttr.Credential = &syscall.Credential{}
 		}
 	}
-	cmd.Env = append(cmd.Env, t)
 	cmd.SysProcAttr.Cloneflags = cloneFlags
+
+	// set init process environment
+	env := []string{"_LIBCONTAINER_INITTYPE=standard"}
+	var joinNamespaces configs.Namespaces
+	var nsPaths []string
+	var doClone bool
+	for _, ns := range c.config.Namespaces {
+		if ns.Path != "" {
+			joinNamespaces = append(joinNamespaces, ns)
+			nsPaths = append(nsPaths, ns.Path)
+			if ns.Type == configs.NEWPID {
+				doClone = true
+			}
+		}
+	}
+	if len(nsPaths) > 0 {
+		env = append(env, fmt.Sprintf("_LIBCONTAINER_NSPATH=%s",
+			strings.Join(nsPaths, ",")))
+	}
+	if doClone {
+		env = append(env, "_LIBCONTAINER_DOCLONE=true")
+	}
+	cmd.Env = append(cmd.Env, env...)
+
 	return &initProcess{
-		cmd:        cmd,
-		childPipe:  childPipe,
-		parentPipe: parentPipe,
-		manager:    c.cgroupManager,
-		config:     c.newInitConfig(p),
+		cmd:            cmd,
+		childPipe:      childPipe,
+		parentPipe:     parentPipe,
+		manager:        c.cgroupManager,
+		config:         c.newInitConfig(p),
+		joinNamespaces: joinNamespaces,
+		doClone:        doClone,
 	}, nil
 }
 
 func (c *linuxContainer) newSetnsProcess(p *Process, cmd *exec.Cmd, parentPipe, childPipe *os.File) *setnsProcess {
+	var nsPaths []string
+	for _, ns := range []string{"ipc", "uts", "net", "pid", "mnt"} {
+		path := fmt.Sprintf("/proc/%d/ns/%s", c.initProcess.pid(), ns)
```

Comment:
in `orderNamespacePaths`, we check for Lstat of the namespaces and fail if it doesnt exist.